### ISSUE TITLE
Failing when using using another Pan Gesture

### ISSUE THREAD

**Issue by WilliamHua** [maxep/MXSegmentedPager#23]
I copied and pasted the code from MXParallaxViewController into a file (I called it SampleViewPager) in my project. 

I then hooked it up to my REFrostedViewController (it is a drawer library I use which uses a panGestureRecognizer to slide the side menu in and out). 

This causes it to crash at removeObserverFromView, specifically when it calls [view removeObserver: self forKeyPath:kContentOffsetKeyPath context:kMXScrollViewKVOContext];

It only crashes when the app loads and I try to scroll something within the tableview / webview / text view

**Comment by WilliamHua** [comment]
I prevented this issue by commenting a few lines in MXSegmentedPager+ParallaxHeader.m:

```

- (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
    UIView<MXPageProtocol> *page = (id) self.segmentedPager.selectedPage;
    BOOL shouldScroll = YES;
    if ([page respondsToSelector:@selector(segmentedPager:shouldScrollWithView:)]) {
        shouldScroll = [page segmentedPager:self.segmentedPager shouldScrollWithView:otherGestureRecognizer.view];
    }
    if (shouldScroll) {
//        [self removeObserverFromView:otherGestureRecognizer.view];
//        [self addObserverToView:otherGestureRecognizer.view];
    }
    return shouldScroll;
}
```

**Comment by maxep** [comment]
Hi @WilliamHua,

Have you try the latest code from the repo ? I have made several changes on this part.

If it's not working, you can use the MXPageProtocol in your pages (by subclassing or categorize it) to prevent the segmented pager from recognize the gesture of the REFrostedViewController.

**Comment by WilliamHua** [comment]
I'll try that right now, I used the versions I received from the pod

**Comment by WilliamHua** [comment]
The bug I had before is gone but I'm having a different issue (that I don't believe to be due to REFrostedViewController). 

```
    UIStoryboard *sb = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    SampleListItem *sample = [sb instantiateViewControllerWithIdentifier:@"SampleListItem"];
    sample.customList = self.lists[0]; // assigning some data I downloaded
    return [@[self.tableView, sample.view] objectAtIndex:index];
```

The self.tableView is the default one from ParallaxHeader. The issue here is similar; the data shows up and I can swipe to the tableView. However when I try to scroll through the TableView (SampleListItem), it crashes and I can't figure out why. 

I tried this with my original ViewController and it didn't work, so I made a new SampleListItem (it's the same thing with a lot of code removed) and I'm having the same issue. The new View Controller (SampleListItem) is just a View Controller with UITableViewDataSource methods implemented. It works by itself, and it also works in a UINavigationBar or embedded in a different ViewPager library. Not sure what could be the issue.

**Comment by maxep** [comment]
Because your 'sample' instance goes out the scope and it's not retain. you should save it somehow.
If your MXSegmentedPager's delegate is a UIViewController, you can use the `addChildViewController:` method like :

``` objective-c
UIStoryboard *sb = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
SampleListItem *sample = [sb instantiateViewControllerWithIdentifier:@"SampleListItem"];
sample.customList = self.lists[0]; // assigning some data I downloaded

[self addChildViewController:sample];
[sample didMoveToParentViewController:self];

return [@[self.tableView, sample.view] objectAtIndex:index];
```

Your other ViewPager lib might be a view-controller and retain it by itself, as the UINavigationViewController. If you want to use the MXSegmentedPager with only view-controllers as pages, you can use the MXSegmentedPagerController and override the `segmentedPager:viewControllerForPageAtIndex:` to specify your pages.

**Comment by WilliamHua** [comment]
You're right that was the issue, thank you so much!

**Comment by maxep** [comment]
You are welcome. FYI, I've released the v1.2 today_
